src/cljs/main/quil_site/examples/floating_graph.cljs

```diff
@@ -119,110 +117,80 @@
                     :search-type search-type
                     :collection (into (search-type collections-init-by-search-type) [starting-node-key])}})) ; use the appropriate collection type for the search type
 
-(defn setup 
+(defn setup
   "quil setup step"
-  [] 
-  ;; don't change :dfs/:bfs color names as they are equivalent to search-states's search-type keywords
-  (def colors {:dfs(q/color 220 255 255)
-               :bfs(q/color 225 150 225)
-               :visiting (q/color 0)
-               :edge (q/color 240 150)
-               :untouched (q/color 240 150)})
-
+  []
   (q/frame-rate frame-rate)
 
   (q/color-mode :rgb)
 
-  ;; create initial nodes and edges ~randomly 
-  (let [nodes (loop [nodes {} amt-to-add node-count]
-                (let [label (keyword (str "n" amt-to-add))
-                      node (node label
-                                 {:x (rand-int (q/width))
-                                  :y (rand-int (q/height))}
-                                 :untouched
-                                 (+ size-min (rand-int (- size-max size-min)))
-                                 {:x (rand-bi-dir directional-speed-max)
-                                  :y (rand-bi-dir directional-speed-max)
-                                  :s (rand-bi-dir size-acceleration)}
-                                 '())
-                      nodes (assoc nodes label node)]
-                  (if (> amt-to-add 0) 
-                    (recur nodes (dec amt-to-add))
-                    nodes)))
-        nodes (loop [nodes nodes amt-to-add edge-count]
-                (let [ks (keys nodes)
-                      k1 (rand-nth ks)
-                      ks (into [] (disj (into #{} ks) k1))
-                      k2 (rand-nth ks)
-                      nodes (add-edge nodes #{k1 k2})] ; edge might already exist, but whatever
-                  (if (> amt-to-add 0)
-                    (recur nodes (dec amt-to-add))
-                    nodes)))]
-    (infer-new-search-state :dfs nodes)))
+  ;; create initial nodes and edges ~randomly
+  (let [labels (map #(keyword (str "n" %))
+                    (range node-count))
+        nodes (into {} (for [label labels]
+                         [label (create-random-node label)]))
+        create-random-edge #(take 2 (shuffle labels))
+        edges (repeatedly edge-count create-random-edge)
+        nodes (reduce add-edge nodes edges)
+        state (infer-new-search-state :dfs nodes)]
+    (assoc state
+           ;; don't change :dfs/:bfs color names as they are equivalent
+           ;; to search-states's search-type keywords
+           :colors {:dfs (q/color 220 255 255)
+                    :bfs (q/color 225 150 225)
+                    :visiting (q/color 0)
+                    :edge (q/color 240 150)
+                    :untouched (q/color 240 150)})))
```

Comment:
-                    :untouched (q/color 240 150)})))
+                    :untouched (q/color 240 150)}
+           :edges (extract-edges (vals nodes)))))

(Extract edges only in setup, combined with annotation below. Using extract-nodes, even though another option is to make make a set out of the previously created list of edges, because it seems more appropriate for the edge set to be information gathered directly from the graph data structure, rather than an independent set of data.)